### PR TITLE
Implement the cGroups sidecar correctly

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @csrwng

### CODE REVIEW

File: cluster/ci/origin/config/tests.yaml [hunk 1]
```diff
@@ -142,11 +142,39 @@ objects:
         name: logging
       - mountPath: /var/run/secrets/gce
         name: gce
+    - command:
+      - /bin/bash
+      args:
+      - -c
+      - 'while true; do uid="$( oc get pod "${POD_NAME}" -n "${POD_NAMESPACE}" -o jsonpath={.metadata.uid} )"; qosClass="$( oc get pod "${POD_NAME}" -n "${POD_NAMESPACE}" -o jsonpath={.status.qosClass} )"; containerID="$( oc get pod "${POD_NAME}" -n "${POD_NAMESPACE}" -o jsonpath="{.status.containerStatuses[?(@.name==\"test\")].containerID}" )"; cat "/sys/fs/cgroup/memory/kubepods.slice/kubepods-${qosClass,,}.slice/kubepods-${qosClass,,}-pod${uid//-/_}.slice/docker-${containerID#docker://}.scope/memory.max_usage_in_bytes"; if [[ -z "$( oc get pod "${POD_NAME}" -n "${POD_NAMESPACE}" -o jsonpath="{.status.containerStatuses[?(@.name==\"test\")].state.terminated}" )" ]]; then sleep 10; else break; fi; done;'
```
Comment (stevekuznetsov) [inline cluster/ci/origin/config/tests.yaml]: Forgive me father for I have sinned

Comment (csrwng) [reply cluster/ci/origin/config/tests.yaml]: indeed

File: cluster/ci/origin/config/tests.yaml [hunk 2]
```diff
@@ -142,11 +142,39 @@ objects:
...
+      - name: POD_NAMESPACE
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.namespace
     dnsPolicy: ClusterFirst
     imagePullSecrets:
     - name: "${PULL_SECRET_NAME}"
     restartPolicy: Never
     terminationGracePeriodSeconds: 5
+    serviceAccount: cgroups-reader
```
Comment (csrwng) [inline cluster/ci/origin/config/tests.yaml]: you only need serviceAccountName

File: cluster/ci/origin/config/tests.yaml [hunk 3]
```diff
@@ -142,11 +142,39 @@ objects:
...
+      imagePullPolicy: IfNotPresent
+      name: cgroups
+      securityContext:
+        privileged: true
+        runAsUser: 0
+      volumeMounts:
+      - mountPath: /sys/fs/cgroup/memory
+        name: cgroupfs
+        readOnly: true
+      terminationMessagePath: /tmp/steve
```
Comment (csrwng) [inline cluster/ci/origin/config/tests.yaml]: on purpose ?

Comment (stevekuznetsov) [reply cluster/ci/origin/config/tests.yaml]: whoops. holdover

**Comment by stevekuznetsov** [comment]
@csrwng updated

**Pull request merged**